pkg/apis/core/validation/validation.go

```diff
@@ -4043,6 +4043,31 @@ func ValidateService(service *core.Service) field.ErrorList {
 		ports[key] = true
 	}
 
+	// Validate TopologyKeys
+	if len(service.Spec.TopologyKeys) > 0 {
+		topoKeys := make(sets.String)
+		topoPath := specPath.Child("topologyKeys")
+		if len(service.Spec.TopologyKeys) > core.MaxServiceTopologyKeys {
+			allErrs = append(allErrs, field.TooMany(topoPath, len(service.Spec.TopologyKeys), core.MaxServiceTopologyKeys))
+		}
+		for i, key := range service.Spec.TopologyKeys {
+			keyPath := topoPath.Index(i)
+			// "" must be the last value specified
+			if key == v1.TopologyKeyAny && i != len(service.Spec.TopologyKeys)-1 {
+				allErrs = append(allErrs, field.Invalid(keyPath, key, "must be the last value specified"))
```

Comment:
The error message does not say `what` must be the last key.

Reply:
done